Modified code to allow for stripping curves in Discount Curve that ha…

tleitch opened the pull request:
…vefixed rate coupon frequency other than annual, fixed day count convetions other than Thirty360, and floating frequency reset other than 6 months. These are critical for adding basis curve impact for discounting.

tleitch commented:
Dirk,
Here are changes in pull. Sorry I included Rcpp export as it blows the files up, but really there are 4 code files changes, discount.R (the calling function) discount.cpp (the RcPP called function) and curves.cpp as well as Rquanlib_internal.h . The three variables are a daycount paramter, a fixed coupon frequency parameter, and a floating reset frequency parameter. They are utilized in curves.cpp call to Quantlib.

I left the original rate helper and created a new definition for the curve stripper as the Bermudan option module expects the original version and I did not see a reason to mod the swaption code at this time.

eddelbuettel commented on man/DiscountCurve.Rd:
```diff
@@ -59,6 +59,11 @@ DiscountCurve(params, tsQuotes, times)
     maturity of the instruments used for calibration (no extrapolation).
     }
+  \item{cpnParams}{A list specifying the \code{dayCounter} the day count convention for the 
+   fixed leg (default is Thirty360), and  \code{freq}, fixed coupon frequecny (defualt is Annua).
```
Annual, not Annua.

eddelbuettel commented:
That looks good at a first (uncaffeniated) glance. Sadly a lot of changes appear to just be whitespace which makes it a little noisy -- maybe we still have tabs in the files from way back when.  There is also rquantlib-tl.Rproj which simply does not belong.  I can clean that up, but if you could just 'git rm' it I'd appreciate it.  

Also, leaving indentation of existing files rather than changing it is considered polite. I prefer 4 spaces just like QuantLib. See [here](http://quantlib.org/style.shtml) and I also follow the R style (described in one of the manuals).

ChangeLog and NEWS entries would also be nice. I can add those.

But the bigger issue is:   it is a clean extension which passes unit tests.  Nice!  I'll look more closely when I have a moment.

tleitch commented:
Thanks! One thing you should be made aware of is the fact the unit test do not exercise (or exorcise) the fixed income routines. The "tests" directory has a master script Rquantlib.R and a matching output that I checked against, but while that script has a non-flat term structure, it isn't utilized. I checked old released version versus new version with the term structure in the file. To use the term structure in the file with the enclosed params you also need to set the evaluation date properly with setEvalutaionDate or the interpolation method will fail.

I'll set the spacing in my various RStudio's I'm using to develop (centos,osx,ubuntu and windows)

eddelbuettel commented:
Look in inst/unitTests/ for the testing framework. It uses RUnit. I can walk you through.

Thanks for switching to 4 spaces. I think if you used the existing Rproj file it may work too...

tleitch commented:
I removed the project file and fixed the typo and pushed, do I execute
another pull?

From:  Dirk Eddelbuettel <notifications@github.com>
Reply-To:  eddelbuettel/rquantlib
<reply+007042a2717f699a853e99e6b811c546286adc2321cbd96692cf000000011303bfe19
2a169ce087327ae@reply.github.com>
Date:  Friday, March 18, 2016 at 9:08 AM
To:  eddelbuettel/rquantlib <rquantlib@noreply.github.com>
Cc:  terry leitch <tleitch@uchicago.edu>
Subject:  Re: [rquantlib] Modified code to allow for stripping curves in
Discount Curve that ha (#31)


Look in inst/unitTests/ for the testing framework. It uses RUnit. I can walk
you through.

Thanks for switching to 4 spaces. I think if you used the existing Rproj
file it may work too...


You are receiving this because you authored the thread.
Reply to this email directly or view it on GitHub
<https://github.com/eddelbuettel/rquantlib/pull/31#issuecomment-198347943>

eddelbuettel commented:
No, GitHub is smarter than that.  The page at https://github.com/eddelbuettel/rquantlib/pull/31 lists this discussion and also the fact that there are now two commits.

eddelbuettel commented:
Ok, I'll merge and post-process, restablished standard indentation and what not.

eddelbuettel commented:
Hm, why where the commits made by a Mark at a domain `frutig.com` which points to Wilmette, IL? 

The file [ChangeLog](https://github.com/eddelbuettel/rquantlib/blob/master/ChangeLog) needs a name and email.  I will put yours in for now.  Should that be changed?

tleitch commented:
not sure why his domain shows up, he worked on another project with me last year and we share some repositories on github. yes, use my name and email and I will look into and fix that going forward.

Sent from my iPhone

> On Mar 19, 2016, at 8:34 AM, Dirk Eddelbuettel <notifications@github.com> wrote:
> 
> Hm, why where the commits made by a Mark at a domain frutig.com which points to Wilmette, IL?
> 
> The file ChangeLog needs a name and email. I will put yours in for now. Should that be changed?
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
>

eddelbuettel commented on inst/include/rquantlib_internal.h:
```diff
@@ -166,15 +169,17 @@ QuantLib::DateGeneration::Rule getDateGenerationRule(const double n);
 boost::shared_ptr<QuantLib::FixedRateBond> getFixedRateBond(Rcpp::List bondparam, std::vector<double> ratesVec, Rcpp::List scheduleparam);
-boost::shared_ptr<QuantLib::IborIndex> getIborIndex(Rcpp::List index, const QuantLib::Date today);
+//boost::shared_ptr<QuantLib::IborIndex> getIborIndex(Rcpp::List index, const QuantLib::Date today);////******
+/*boost::shared_ptr<QuantLib::IborIndex> getIborIndex(std:string type);////******/
```
What are you trying to say here with `//******` ?  To be revisited?

It is a real pain we don't have clean diff's.  I am restoring whitespace to what it was so the diff between the _next_ and the _previous_ diff will be more meaningful.

eddelbuettel commented:
It will be an issue related to your github issue. You may want to look into that going forward. 

Almost done cleaning.  Was a bit more painful than it needed to be, but now (at last ;-) I see your changes more clearly and they are nice and clean.   We will get coding and indentation style etc sorted out.

eddelbuettel commented:
I pushed a new branch which I will probably merge into master shortly too.  If you set my repository up as a remote 'parentrepo' (eg via something like

```
git remote add --track master parentrepo https://github.com/eddelbuettel/rquantlib.git
```

then 

```
git pull --all
git merge parentrepo/master
```

will sync your fork.  Right now you'd have to merge with the branch that has my changes.

eddelbuettel commented:
Master now has my #32 following this #31.   

Onwards ...    ;-)
